sonic-xcvrd/xcvrd/xcvrd.py

```diff
@@ -1432,6 +1464,20 @@ def task_worker(self):
                         self.port_dict[lport]['cmis_state'] = self.CMIS_STATE_READY
                         continue
 
+                    appl = self.get_cmis_application_desired(api, host_lane_count, host_speed)
+                    if appl < 1:
+                        self.log_error("{}: no suitable app for the port".format(lport))
+                        self.port_dict[lport]['cmis_state'] = self.CMIS_STATE_FAILED
+                        continue
+
+                    host_lanes_mask = self.get_cmis_host_lanes_mask(api, appl, host_lane_count, channel, speed)
```

Comment:
@mihirpat1  can you check if the fn get_cmis_application_desired() and get_cmis_host_lanes_mask() are getting executed during every state machine transition? which is unnecessary? may be we should call once inside the CMIS_STATE_INSERTED ?

Reply:
I am now calling get_cmis_application_desired and get_cmis_host_lanes_mask only in CMIS_STATE_INSERTED. In other states, I am using the value from port_dict dictionary.
I am also making appl and host_lanes_mask to 0 while handling CMIS_STATE_REMOVED.